signer/tests/integration/contracts.rs

```diff
@@ -318,3 +317,62 @@ async fn is_deposit_completed_works() {
 
     assert_eq!(response, true);
 }
+
+/// This tests whether the `is_withdrawal_completed` function works when
+/// the smart contract has been completed with a rejection.
+#[ignore = "This is an integration test that requires a stacks-node to work"]
+#[tokio::test]
+async fn is_withdrawal_completed_rejection_works() {
+    let stacks_client = stacks_client();
+    let stacks_client = stacks_client.get_client();
+    let rpc = {
+        let username = regtest::BITCOIN_CORE_RPC_USERNAME.to_string();
+        let password = regtest::BITCOIN_CORE_RPC_PASSWORD.to_string();
+        let auth = bitcoincore_rpc::Auth::UserPass(username, password);
+        bitcoincore_rpc::Client::new("http://localhost:18443", auth).unwrap()
+    };
+
+    let signers = deploy_smart_contracts().await;
+    let chain_tip_info = rpc.get_chain_tips().unwrap().pop().unwrap();
+
+    let txid: BitcoinTxId = Faker.fake_with_rng(&mut rand::rngs::OsRng);
+
+    let mint_sbtc = ContractCallWrapper(CompleteDepositV1 {
+        outpoint: bitcoin::OutPoint::new(txid.into(), 0),
+        amount: 123654789,
+        recipient: PrincipalData::from(*testing::wallet::WALLET.0.address()),
+        deployer: *testing::wallet::WALLET.0.address(),
+        sweep_txid: BitcoinTxId::from([0; 32]),
+        sweep_block_hash: chain_tip_info.hash.into(),
+        sweep_block_height: chain_tip_info.height,
+    });
+
+    let _txid = signers.sign_and_submit(&mint_sbtc).await;
+
+    let start_withdrawal = ContractCallWrapper(InitiateWithdrawalRequest {
+        amount: 22500,
+        recipient: (0x00, vec![0; 20]),
+        max_fee: 3000,
+        deployer: *testing::wallet::WALLET.0.address(),
+    });
+
+    let _txid = signers.sign_and_submit(&start_withdrawal).await;
+
+    let reject_withdrawal = ContractCallWrapper(RejectWithdrawalV1 {
+        request_id: 1,
+        signer_bitmap: BitArray::ZERO,
+        deployer: *testing::wallet::WALLET.0.address(),
+    });
+
+    let _txid = signers.sign_and_submit(&reject_withdrawal).await;
+
+    // Wait for confirmation. The stacks node takes it's sweet time
+    // confirming transactions.
+    tokio::time::sleep(Duration::from_secs(5)).await;
+
+    let response = stacks_client
+        .is_withdrawal_completed(signers.wallet.address(), 1)
+        .await;
```

Comment:
Can we also check that this returns false before the reject call?

Reply:
Done in https://github.com/stacks-network/sbtc/pull/1377/commits/251001039369ef9392f1e4336c9a397d38b3ee97